src/Microsoft.Azure.WebJobs.Host/Scale/ProcessMonitor.cs

```diff
@@ -191,5 +196,59 @@ public void Dispose()
         {
             Dispose(true);
         }
+
+        /// <summary>
+        /// Interface over a system Process. Useful for mock testing because
+        /// Process isn't mockable.
+        /// </summary>
+        public interface IProcess
+        {
+            Process Inner { get; }
+            int Id { get; }
+            bool HasExited { get; }
+            TimeSpan TotalProcessorTime { get; }
+            long PrivateMemoryBytes { get; }
+            void Refresh();
+        }
+
+        private class ProcessWrapper : IProcess
+        {
+            private readonly Process _process;
+
+            public ProcessWrapper(Process process)
+            {
+                _process = process ?? throw new ArgumentNullException(nameof(process));
+            }
+
+            public Process Inner => _process;
+
+            public int Id => _process.Id;
+
+            public bool HasExited
+            {
+                get
+                {
+                    try
+                    {
+                        return _process.HasExited;
+                    }
+                    catch
+                    {
+                        // When the process has been Closed/Disposed
+                        // Process members like HasExited will throw
+                        return true;
```

Comment:
We can't check if its disposed, instead of catching the exception?

Reply:
The IDisposable interface doesn't have an "IsDisposed".